compiler/src/main/java/io/neow3j/compiler/ManifestBuilder.java

```diff
@@ -106,17 +110,92 @@ private static ContractABI buildABI(NeoModule neoModule) {
+                            Collections.singletonList("*"));
+                    return Collections.singletonList(contractPermission);
+                });
+    }
+
+    private static List<String> buildTrusts(ClassNode asmClass) {
+        return getAnnotationNode(asmClass, Trusts.class)
+                .map(ManifestBuilder::getContractTrusts)
+                .orElse(new ArrayList<>());
+    }
+
+    private static Optional<List<String>> transformAnnotationNodeStringValue(
+            AnnotationNode annotationNode) {
+
+        return ofNullable(annotationNode)
+                .map(ann -> {
+                    List<String> values = new ArrayList<>();
+                    for (Object value : (List<?>) ann.values.get(1)) {
+                        values.add((String) value);
+                    }
+                    return values;
+                });
+    }
+
+    private static List<ContractPermission> getContractPermissions(AnnotationNode ann) {
+        List<ContractPermission> permissions = new ArrayList<>();
+        for (AnnotationNode permission : (List<AnnotationNode>) ann.values.get(1)) {
+            int i = permission.values.indexOf("contract");
+            String contract = (String) permission.values.get(i + 1);
+
+            i = permission.values.indexOf("methods");
+            List<String> methods = new ArrayList<>();
+            // if 'methods' is not found, it means we need to add a "wildcard"
+            // to that manifest
+            if (i < 0) {
+                methods.add("*");
+            } else {
+                List<?> methodsValues = (List<?>) permission.values.get(i + 1);
+                // this is required since we want to create an ArrayList of new
+                // String objects, and not rely on what ASM provides us
+                methods.addAll((List<String>) methodsValues);
+            }
+
+            ContractPermission contractPermission = new ContractPermission(contract, methods);
+            permissions.add(contractPermission);
+        }
+        return permissions;
+    }
+
+    private static List<ContractGroup> getContractGroups(AnnotationNode ann) {
+        List<ContractGroup> groups = new ArrayList<>();
+        for (AnnotationNode group : (List<AnnotationNode>) ann.values.get(1)) {
+            int i = group.values.indexOf("pubKey");
+            String pubkey = (String) group.values.get(i + 1);
+            i = group.values.indexOf("signature");
+            String signature = (String) group.values.get(i + 1);
+            ContractGroup cgManifest = new ContractGroup(pubkey, signature);
+            groups.add(cgManifest);
         }
-        AnnotationNode ann = annotationNode.get();
-        List<String> standards = new ArrayList<>();
-        for (Object standard : (List<?>) ann.values.get(1)) {
-            standards.add((String) standard);
+        return groups;
+    }
+
+    private static List<String> getContractTrusts(AnnotationNode ann) {
```

Comment:
We could enforce rules on the annotation values here.
The value has to be either a 33 (public key) or 20 bytes long (script hash) hex string.

Reply:
Yes, good point. I will do that.

Reply:
implemented.